### PR TITLE
Fix judgements not scaling with circle size

### LINKED ISSUE #3195: Judgements do not scale with circle size
The ingame judgements should scale with CS so their size fits the size of the clicked circle. Currently every judgement has the same set size which looks a bit weird for maps with high/low CS values.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/61903134-judgements-do-not-scale-with-circle-size?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by qwewqa**
Fixes #3195

### CODE REVIEW

File: osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs [hunk 1]
```diff
@@ -69,10 +72,13 @@ private void onNewResult(DrawableHitObject judgedObject, JudgementResult result)
...
                 return;
 
+            var explosionBaseSize = rulesetContainer.Beatmap.BeatmapInfo.BaseDifficulty.CircleSize;
+
             DrawableOsuJudgement explosion = new DrawableOsuJudgement(result, judgedObject)
             {
                 Origin = Anchor.Centre,
-                Position = ((OsuHitObject)judgedObject.HitObject).StackedEndPosition
+                Position = ((OsuHitObject)judgedObject.HitObject).StackedEndPosition,
+                Scale = new Vector2((1.0f - 0.7f * (explosionBaseSize - 5) / 5) / 2 * 1.65f)
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs]: This should probably just come from the `judgedObject`'s scale?

**Review by peppy — DISMISSED** [review]

File: osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs (hunk 1 above)
Comment (qwewqa) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs]: Sometimes it comes from the slider which always has a (1, 1) scale

Comment (peppy) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs]: see `OsuHitObject.Scale`. ie. `judgedObject.HitObject.Scale`

**Review by smoogipoo — APPROVED** [review]
This seems to look okay.

File: osu.Game/Rulesets/Judgements/DrawableJudgement.cs [hunk 2]
```diff
@@ -65,22 +65,24 @@ protected override void LoadComplete()
...
+            var origScale = Scale;
+
             switch (Result.Type)
             {
                 case HitResult.None:
                     break;
                 case HitResult.Miss:
-                    this.ScaleTo(1.6f);
-                    this.ScaleTo(1, 100, Easing.In);
+                    this.ScaleTo(origScale * 1.6f);
```
Comment (peppy) [inline osu.Game/Rulesets/Judgements/DrawableJudgement.cs]: probably shouldn't be adjusting ones' own scale, but i'll let this slide this time

Comment (qwewqa) [reply osu.Game/Rulesets/Judgements/DrawableJudgement.cs]: What would be better in the future, just wondering

**Pull request merged**